### DISCUSSION TITLE
can next/navigation only be used by components that are imported into the app folder?

### DISCUSSION THREAD

**Discussion by ethanryan** [vercel/next.js#50279] [App Router] [answered]
possibly related issue: https://github.com/vercel/next.js/issues/46360

Begun migrating from "/pages" to "/app" folder.

Starting with a very simple example page that renders our header and footer. The header previously imported { useRouter } from 'next/router', I updated it to import { usePathname } from 'next/navigation'.

Header component used:

```
import { useRouter } from 'next/router'
```

And now uses:

```
import { usePathname } from 'next/navigation'
```

Build is failing after updating that component, and importing it in the example page layout.

Component uses typescript. 

Build succeeds if that component is not imported into new example page. 

Also tried wrapping component in Suspense, but build still fails.

Build fails with: 

```
Collecting page data

"false" = namespaces;

› Build error occurred
Error: Failed to collect page data for /example
```

**Comment by ciruz** [comment, marked as answer]
You can only use the `useRouter` and/or `usePathname` hooks in [Client Components](https://nextjs.org/docs/getting-started/react-essentials#client-components). By default all components are Server Components, where these hooks are not available.

You need to define Server Components as Client Components with `"use client"` on the begining of the file, when you want to use these hooks in the file.

```tsx
// header.tsx
"use client"

import { useRouter } from 'next/navigation';

...
```

> Migrating from the pages directory:

> - The new `useRouter` hook should be imported from `next/navigation` and not `next/router`
> - The `pathname` string has been removed and is replaced by [`usePathname()`](https://nextjs.org/docs/app/api-reference/functions/use-pathname)
> - The `query` object has been removed and is replaced by [`useSearchParams()`](https://nextjs.org/docs/app/api-reference/functions/use-search-params)
> - router.events is not currently supported. [See below.](https://nextjs.org/docs/app/api-reference/functions/use-router#router-events)

Also checkout the [Migration Guide](https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration) from Pages to App directory.

**Reply by ethanryan** [reply]
OK thanks @ciruz, so we can use 'next/navigation' in client components, whether they are rendered by parent components in the "/app" folder or in the "/pages" folder.

The migration guide lists "migrating routing hooks" as step 5, but it seems like it should be step 1. 

https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration#step-5-migrating-routing-hooks

Migrating all those routing hooks first should make the other steps easier.
